plugins/outputs/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,127 @@
+package opentelemetry
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/influxdata/influxdb-observability/common"
+	"github.com/influxdata/influxdb-observability/influx2otel"
+	otlpcollectormetrics "github.com/influxdata/influxdb-observability/otlp/collector/metrics/v1"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"google.golang.org/grpc"
+)
+
+type OpenTelemetry struct {
+	ServiceAddress string `toml:"service_address"`
+	MetricsSchema  string `toml:"metrics_schema"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	metricsConverter     *influx2otel.LineProtocolToOtelMetrics
+	grpcClientConn       *grpc.ClientConn
+	metricsServiceClient otlpcollectormetrics.MetricsServiceClient
+}
+
+func newOpenTelemetry() *OpenTelemetry {
+	return &OpenTelemetry{
+		ServiceAddress: "localhost:4317",
+		MetricsSchema:  "prometheus-v1",
+	}
+}
+
+const sampleConfig = `
+  ## Override the OpenTelemetry gRPC service address:port
+  # service_address = "localhost:4317"
+
+  ## Override the default request timeout
+  # timeout = "5s"
+
+  ## Select a schema for metrics: prometheus-v1 or prometheus-v2
+  ## For more information about the alternatives, read the Prometheus input
+  ## plugin notes.
+  # metrics_schema = "prometheus-v1"
+`
+
+func (o *OpenTelemetry) SampleConfig() string {
+	return sampleConfig
+}
+
+func (o *OpenTelemetry) Description() string {
+	return "Send OpenTelemetry metrics over gRPC"
+}
+
+var metricsSchemata = map[string]common.MetricsSchema{
+	"prometheus-v1": common.MetricsSchemaTelegrafPrometheusV1,
+	"prometheus-v2": common.MetricsSchemaTelegrafPrometheusV2,
+}
+
+func (o *OpenTelemetry) Connect() error {
+	logger := &otelLogger{o.Log}
+	ms, found := metricsSchemata[o.MetricsSchema]
+	if !found {
+		return fmt.Errorf("schema '%s' not recognized", o.MetricsSchema)
+	}
+
+	metricsConverter, err := influx2otel.NewLineProtocolToOtelMetrics(logger, ms)
+	if err != nil {
+		return err
+	}
+
+	grpcClientConn, err := grpc.Dial(o.ServiceAddress, grpc.WithInsecure())
```

Comment:
Using `WithInsecure` here would prevent anyone from using a secure connection, would it be possible to add support for configuring TLS as well? Something like this would be great: https://github.com/influxdata/telegraf/pull/9109/files#diff-9286694f607c96eb8bc3d8f4e1c493a9931f543b009ef2538a03c63dd680418cR1268-R1273

Reply:
agreed. see our `tls.ClientConfig`